### PR TITLE
[NTUSER] IntSetTimer(): Update HintIndex on each call (update)

### PR DESCRIPTION
**Pull request by turican0**
## Purpose

Based on the [Doug-Lyons](https://github.com/Doug-Lyons) test, I found that my previous fix stopped working partially. Or rather, it would only work until the 32767 indexes were exhausted. It seems to me that the behavior of the bitfield has changed, because when I published the previous patch, it passed my tests.

Previous patch versions: #7087

## Proposed changes

- Bit array generates free ID cyclically, in the previous code after 32767 indexes expired the same index was returned, because of this the previous fix would stop working after expiration, so I changed the logic of calculating the next index.
- I changed the index range to 256-32767 to match Windows, indexes 0-255 can theoretically be used as fixed for system purposes.

### CODE REVIEW

File: win32ss/user/ntuser/timer.c
Comment (HBelusca) [inline win32ss/user/ntuser/timer.c:22]:
```suggestion
/* These values give timer IDs [256,32767], same as on Windows */
#define NUM_WINDOW_MAX_TIMERS   (32768 - 1)
#define NUM_WINDOW_LESS_TIMERS  (32768 - 256)
```
That `NUM_WINDOW_MAX_TIMERS` may be named instead something like `MAX_WINDOW_TIMER_ID` if this value is really a particular ID value, not a number of timers.

**Comment by Doug-Lyons** [comment]
I believe that this PR should be considered superseded by Timo's PR #7280 for now. It can be re-considered once his PR is resolved. Thanks for all of your efforts here, @turican0.

**Comment by turican0** [comment]
Ok, if I missed it, let me know when #7277 is in the main thread and I'll do a new PR then.

**Comment by binarymaster** [comment]
While this PR is open, you will be able to rebase commits and force-push to it, if it gets any conflicts with master branch.

**Comment by turican0** [comment]
I'll try, but I'm not very proficient in git. Anyway, I have to wait for merge #7280.

**Comment by turican0** [comment]
@binarymaster @Doug-Lyons @HBelusca Hi, I've updated my PR #7277 so that it can be merged after the changes in #7280.

**Review by HBelusca — APPROVED** [review]
lgtm.
@tkreuzer Could you review this PR, since you've recently fixed some 64-bit portability problems in this region of code?

File: win32ss/user/ntuser/timer.c
Comment (tkreuzer) [inline win32ss/user/ntuser/timer.c:21]: That name doesn't really make sense. Use something like `MAX_WINDOW_LESS_TIMER_ID`

**Review by tkreuzer — APPROVED** [review]
Technically lgtm, just improve that name.

**Review by Doug-Lyons — APPROVED** [review]

File: win32ss/user/ntuser/timer.c
Comment (binarymaster) [inline win32ss/user/ntuser/timer.c:22]: It seems you renamed another define, not the one that was requested to rename by Timo 🤔

Comment (tkreuzer) [reply win32ss/user/ntuser/timer.c:22]: yeah

Comment (turican0) [reply win32ss/user/ntuser/timer.c]: Sorry, I didn't realize it was about the first variable :) I've edited it.

**Pull request merged**
